frontends/main/src/app/providers.tsx

```diff
@@ -18,4 +31,12 @@ export default function Providers({ children }: { children: React.ReactNode }) {
       </NextJsAppRouterCacheProvider>
     </QueryClientProvider>
   )
+
+  return apiKey.length > 0 ? (
+    <PostHogProvider apiKey={apiKey} options={postHogOptions}>
+      {interiorElements}
+    </PostHogProvider>
+  ) : (
+    interiorElements
+  )
```

Comment:
What would you think of doing something like
```tsx
const ConfiguredPHProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const apiKey = process.env.NEXT_PUBLIC_POSTHOG_PROJECT_API_KEY || ""

  const apiHost =
    process.env.NEXT_PUBLIC_POSTHOG_API_HOST || "https://us.i.posthog.com"
  const featureFlags = JSON.parse(process.env.FEATURE_FLAGS || "")

  const postHogOptions = {
    api_host: apiHost,
    bootstrap: {
      featureFlags: featureFlags,
    },
  }
  if (!apiKey) return children
  return (
    <PostHogProvider apiKey={apiKey} options={postHogOptions}>
      {children}
    </PostHogProvider>
  )
}
```
?

Then `ConfiguredPHProvider` can handle the conditional itself + configuration.

Partly suggesting this to encapsulate it, but partly because we could run into the same situation in future with other providers, and as-structured I think it would get awkward with more than 1.